### DISCUSSION TITLE
Sentencizer - sentence splitting breaks on contiguous speech marked texts

### DISCUSSION THREAD

**Discussion by jokase** [explosion/spaCy#7302] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour

```
>>> import spacy; nlp = spacy.load('en_core_web_trf'); nlp.add_pipe("sentencizer", before="parser")
<spacy.pipeline.sentencizer.Sentencizer object at 0x000001FD816C0340>

>>> print (list(nlp('"sentence one." "sentence two."').sents))
["sentence one." ", sentence two."]

>>> print (list(nlp('"sentence one." "sentence two." "Sentence three."').sents))
["sentence one." ", sentence two." ", Sentence three."]
```

**Without sentencizer**
```
>>> import spacy; nlp = spacy.load('en_core_web_trf');                                             
>>> print (list(nlp('"sentence one." "sentence two."').sents))                          
["sentence one.", "sentence two."]

>>> print (list(nlp('"sentence one." "sentence two." "Sentence three."').sents))        
["sentence one.", "sentence two.", "Sentence three."]
```

## Your Environment

* Operating System: Win10 x64
* Python Version Used: Python 3.9.1 (tags/v3.9.1:1e5d33e, Dec  7 2020, 17:08:21) [MSC v.1927 64 bit (AMD64)] on win32
* spaCy Version Used: 3.0.3
* Environment Information:

**Comment by adrianeboyd** [comment, marked as answer]
This is the expected behavior for the `sentencizer` algorithm -- it's fast but not very sophisticated.

If you have a `parser` in your pipeline, it will provide better sentence boundaries and you most likely don't want the additional `sentencizer` component. The main drawback for the using the `parser` vs. `sentencizer` for sentence boundaries is the speed, but in a `trf` pipeline, this difference is probably not a big factor in the overall speed.

And to clarify in case the combined sentencizer+parser behavior is confusing: the parser respects existing sentence boundaries, so if you run the sentencizer first, the parser will keep all the sentencizer boundaries and create parses that use these boundaries.

**Comment by svlandeg** [comment]
(Maintenance note: this question is well suited for the discussions forum, so I'm moving it there. While it looks like this issue gets closed, in fact the linked discussion thread will remain open to continue the conversation)
